datalad/distribution/create.py

```diff
@@ -261,26 +261,26 @@ def __call__(
         # just discard, we have a new story to tell
         path.pop('message', None)
         if 'parentds' in path:
-            try:
-                subds = next(
-                    sds
-                    for sds in Subdatasets.__call__(
-                        dataset=path['parentds'],
-                        # any known
-                        fulfilled=None,
-                        recursive=False,
-                        result_xfm='paths')
-                    if path_startswith(path['path'], sds)
-                )
+            subs = Subdatasets.__call__(
+                dataset=path['parentds'],
+                # any known
+                fulfilled=None,
+                recursive=False,
+                contains=path['path'],
+                result_xfm='relpaths')
+            if len(subs):
                 path.update({
                     'status': 'error',
                     'message': ('collision with known subdataset %s/ in dataset %s',
-                                relpath(subds, path['parentds']), path['parentds'])})
+                                subs[0], path['parentds'])})
                 yield path
                 return
-            except StopIteration:
-                # all good
-                pass
+
+        # TODO here we need a further test that if force=True, we need to look if
+        # there is a superdataset (regardless of whether we want to create a
+        # subdataset or not), and if that superdataset tracks anything within
+        # this directory -- if so, we need to stop right here and whine, because
+        # the result of creating a repo here will produce an undesired mess
```

Comment:
I think I recall, that it was possible to move anything tracked by a potential superdataset into that newly created subdataset. If the user says "force" shouldn't that be the desired result instead of failing?